x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { LegacyAPICaller } from '../../../../../src/core/server';
+import { CollectorDependencies } from './types';
+import { DetectionsUsage, fetchDetectionsUsage } from './detections';
+
+export type RegisterCollector = (deps: CollectorDependencies) => void;
+export interface UsageData {
+  detections: DetectionsUsage;
+}
+
+export const registerCollector: RegisterCollector = ({ kibanaIndex, ml, usageCollection }) => {
+  if (!usageCollection) {
+    return;
+  }
+
+  const collector = usageCollection.makeUsageCollector<UsageData>({
+    type: 'security_solution',
+    schema: {
+      detections: {
+        detection_rules_custom_enabled: { type: 'number' },
+        detection_rules_custom_disabled: { type: 'number' },
+        detection_rules_elastic_enabled: { type: 'number' },
+        detection_rules_elastic_disabled: { type: 'number' },
+        ml_jobs_custom_enabled: { type: 'number' },
+        ml_jobs_custom_disabled: { type: 'number' },
+        ml_jobs_elastic_enabled: { type: 'number' },
+        ml_jobs_elastic_disabled: { type: 'number' },
+      },
```

Comment:
NIT: Can we use `type: 'long'` instead (or `'float'` if that's the intended format)?
This will help us understand the type of number to be indexed 🙂

Reply:
NIT 2: Have you considered using a tree structure instead?

```JSON
    {
      "properties": {
        "detections": {
          "properties": {
            "detection_rules": {
              "custom": { 
                "enabled": { "type": "long" },
                "disabled": { "type": "long" }
              },
              "elastic": { 
                "enabled": { "type": "long" },
                "disabled": { "type": "long" }
              }
            },
            "ml_jobs": {
              "custom": { 
                "enabled": { "type": "long" },
                "disabled": { "type": "long" }
              },
              "elastic": { 
                "enabled": { "type": "long" },
                "disabled": { "type": "long" }
              }
            }
          }
        }
      }
    }
```